pkg/collector/corechecks/servicediscovery/impl_linux_test.go

```diff
@@ -0,0 +1,354 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package servicediscovery
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/google/go-cmp/cmp"
+	"github.com/prometheus/procfs"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/comp/core/hostname/hostnameinterface"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/mocksender"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/portlist"
+)
+
+type testProc struct {
+	pid     int
+	cmdline []string
+	env     []string
+	cwd     string
+	stat    procfs.ProcStat
+}
+
+func mockProc(
+	ctrl *gomock.Controller,
+	p testProc,
+) proc {
+	m := NewMockproc(ctrl)
+	m.EXPECT().PID().Return(p.pid).AnyTimes()
+	m.EXPECT().CmdLine().Return(p.cmdline, nil).AnyTimes()
+	m.EXPECT().Environ().Return(p.env, nil).AnyTimes()
+	m.EXPECT().Cwd().Return(p.cwd, nil).AnyTimes()
+	m.EXPECT().Stat().Return(p.stat, nil).AnyTimes()
+	return m
+}
+
+func TestServiceDiscoveryCheckLinux(t *testing.T) {
```

Comment:
For some reason, I can't get these tests to run on my Mac. Are there any special instructions with the mock library? I am not a fan of using them.

Reply:
This test file has build tags `//go:build linux` since its using types / functions from `impl_linux.go` which has the same build tags (mockgen should not be the problem since its just autogenerating code).

 I'm running them in a debian VM (a docker container should also work) with the following command:

```
go test -v ./pkg/collector/corechecks/servicediscovery/ -tags=test,linux
```